src/openzeppelin/tests/token/test_erc20.cairo

```diff
@@ -149,17 +167,22 @@ fn test_approve_from_zero() {
 #[should_panic(expected: ('ERC20: approve to 0', ))]
 fn test_approve_to_zero() {
     let mut state = setup();
-    set_caller_address(OWNER());
+    testing::set_caller_address(OWNER());
     ERC20Impl::approve(ref state, Zeroable::zero(), VALUE);
 }
 
 #[test]
 #[available_gas(2000000)]
 fn test__approve() {
     let mut state = setup();
-    set_caller_address(OWNER());
+    testing::set_caller_address(OWNER());
     InternalImpl::_approve(ref state, OWNER(), SPENDER(), VALUE);
 
+    let event = testing::pop_log::<Approval>(ZERO()).unwrap();
+    assert(event.owner == OWNER(), 'Invalid owner');
+    assert(event.spender == SPENDER(), 'Invalid spender');
+    assert(event.value == VALUE, 'Invalid value');
```

Comment:
again following Ownable PR's style, we should abstract these into a util to avoid code repetition while also checking no extra parameters nor events are being fired